model/vm_host.rb

```diff
@@ -254,6 +262,39 @@ def check_pulse(session:, previous_pulse:)
     pulse
   end
 
+  VM_HOST_NVME_DISK_COMMAND = "nvme list"
+  CheckNvmeDisksStatus = lambda do |cmd_output|
+    # Handle different cases based on the command output
+    case cmd_output
+    when /segmentation fault/i
+      "down"
+    # Check if any NVMe disks are listed in the response
+    else
+      if cmd_output.lines.any? { |line| line.start_with?('/dev/nvme') }
+        "up"
+      else
+        "no_nvme_disks"
+      end
+    end
```

Comment:
you can further improve the logic here by looking into `nvme smart-log` output. It gives critical information on `Critical warnings` or `temperature`, we may utilize this information to deduce the state of the disk.

Reply:
Also, when `nvme list` returns the disks, it doesn't necessarily mean that they are connected properly and we can make use of them.

Reply:
Sure, will add.